source/core/types/ut_executable.tpb

```diff
@@ -135,9 +136,14 @@ create or replace type body ut_executable is
 
       l_completed_without_errors := (self.error_stack||self.error_backtrace) is null;
 
-      a_listener.fire_after_event(self.associated_event_name, a_item);
       --listener - after call to executable
+      a_listener.fire_after_event(self.associated_event_name, a_item);
+
+      if l_start_transaction_id != dbms_transaction.local_transaction_id(true) then
```

Comment:
I mean, we need to create transaction at the start.
You're right, we don't need to create it at the end, but do an additional null check.
If you think it should be changed, so that we don't create transaction after execution, can you update the PR?
I'm ok with both solutions.